src/main/java/gift/Service/WishService.java

```diff
@@ -4,39 +4,64 @@
 import gift.Repository.ProductRepository;
 import gift.Repository.WishRepository;
 import jakarta.transaction.Transactional;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
 import org.springframework.stereotype.Service;
 
 import java.util.*;
 
 @Service
 public class WishService {
-    private final WishRepository WishRepository;
+    private final WishRepository wishRepository;
     private final ProductRepository productRepository;
 
 
-    public WishService(WishRepository WishRepository, ProductRepository productRepository) {
-        this.WishRepository = WishRepository;
+    public WishService(WishRepository wishRepository, ProductRepository productRepository) {
+        this.wishRepository = wishRepository;
         this.productRepository = productRepository;
     }
 
 
 
+    @Transactional
     public void addWish(Member member, RequestWishDTO requestWishDTO) {
         Product product = productRepository.findById(requestWishDTO.getProductId())
                 .orElseThrow(()->new NoSuchElementException("매칭되는 물건이 없습니다."));
         Wish wish = new Wish(member, product, requestWishDTO.getCount());
-        WishRepository.save(wish);
+        wishRepository.save(wish);
+    }
+
+    public Page<Wish> getWishList(Member member, int page, int pageSize, String sortField, String sortDir) {
+        Sort sort = sortDir.equalsIgnoreCase(Sort.Direction.ASC.name()) ? Sort.by(sortField).ascending() : Sort.by(sortField).descending();
+        Pageable pageable = PageRequest.of(page, pageSize, sort);
+        Page<Wish> wishListPage= wishRepository.findByMember(member,pageable);
+        return wishListPage;
     }
 
     public List<ResponseWishDTO> getWish(Member member) {
-        return WishRepository.findWishListByMember(member);
+        List<Wish> wishList = wishRepository.findWishListByMember(member);
+        ListIterator<Wish> iterator = wishList.listIterator();
+        List<ResponseWishDTO> responseWishDTOList = new ArrayList<>();
+        while(iterator.hasNext()){
+            Wish wish = iterator.next();
+            responseWishDTOList.add(new ResponseWishDTO(wish.getProduct().getName(), wish.getCount()));
+        }
```

Comment:
```suggestion
        List<ResponseWishDTO> responseWishDTOList = new ArrayList<>();
        for (Wish wish : wishList) {
            responseWishDTOList.add(new ResponseWishDTO(wish.getProduct().getName(), wish.getCount()));
        }
```

`for-each` 문이나 `stream`을 사용하면 조금 더 간결한 표현이 가능하답니다 😄

Reply:
넵 해당사항 반영토록 하겠습니다!